services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImpl.java

```diff
@@ -292,6 +300,7 @@ public News scheduleNews(News news, org.exoplatform.services.security.Identity c
     if (!canScheduleNews(space, currentIdentity)) {
       throw new IllegalArgumentException("User " + currentIdentity.getUserId() + " is not authorized to schedule news");
     }
+    linkoNewsToTargets(news);
```

Comment:
I think that you should call instead removeNewsTargets then setNewsTargets in publishNews method